### PR TITLE
Fundamentals of JS (70%) done

### PR DESCRIPTION
**Pull request by Rolalove**
I worked on the Fundamentals, technically covered 70% of the sub-topic. I could not cover type conversion, (Logical, Bitwise, Null Coalescing, Optional Chaining)  under operator sub-topic.  I should do that latest sunday evening or monday morning. I also want to know if this approach is good. 

#3

### CODE REVIEW

**Comment by Oluwasetemi** [comment]
sorry this is a test comment 2

**Comment by Oluwasetemi** [comment]
total new comment testing new workflows

**Comment by Oluwasetemi** [comment]
any update on this @Rolalove?

**Comment by Oluwasetemi** [comment]
any update on this @Racheal-cloud?

**Comment by calmifytech** [comment]
> any update on this @Rolalove?

Yes , I will be sending a PR today.

File: pages/fundamentals.md [hunk 1]
```diff
@@ -14,29 +14,175 @@ hideInToc: true
...
+---
+
+```js
+//Undefined Datatype
+- Undefined indicate that a variable has not been assigned a value, or not declared at all.
+let age; // the variable age was declared but not assigned a value
+let age = undefined //it also possible to explicitly assign undefined to a variable
+
+//BigInt Datatype 
+- BigINt are used to store big integer values that too big to be represented by normal Javascript Number
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:140]: there is a typo on the BigInt here.

File: pages/fundamentals.md [hunk 2]
```diff
@@ -65,6 +215,457 @@ let isMarried = confirm('Are you married?');
...
+3. Numeric Conversion
+4. Boolean coversion
+
+---
+hideInToc: true
+---
+
+# Operators: `Math`, `Comparison`, `Logical`,` Bitwise`, `Null Coalescing`, `Optional Chaining`
+### The following operators are suported in Js 
+- Additon  ( + )
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:236]: spelling here as well.

File: pages/fundamentals.md [hunk 3]
```diff
@@ -65,6 +215,457 @@ let isMarried = confirm('Are you married?');
...
+}
+```
+
+
+---
+hideInToc: true
+---
+
+- Operators: `Math`, `Comparison`, `Logical`,` Bitwise`, `Null Coalescing`, `Optional Chaining`
+### The following operators are suported in Js
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:312]: Slide 19 is the same as 15

File: pages/fundamentals.md [hunk 4]
```diff
@@ -65,6 +215,457 @@ let isMarried = confirm('Are you married?');
...
+console.log(canVote)
+```
+---
+hideInToc: true
+---
+
+- <span class='text-[#5D8392]'>Switch </span>: A switch statement can replace multiple if checks. it gives a cleaner way of comparing a value with multiple variant
+- Syntax
+```js
+switch (expression)
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:437]: there is a curl braces missing here

File: pages/fundamentals.md [hunk 5]
```diff
@@ -65,6 +215,457 @@ let isMarried = confirm('Are you married?');
...
+    i++;
+}
+```
+- <span class='text-[#5D8392]'>For loop</span>: The syntax is quite simple. it has 3 major part.
+```js
+for (initialization part, condition check, increment part){
+    // loop body
+}
+```
+- Initialization part: is executed (one time) before the excecution of the code block
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:518]: the spelling of execution

File: pages/fundamentals.md [hunk 6]
```diff
@@ -65,6 +215,457 @@ let isMarried = confirm('Are you married?');
...
+let language = 'JavaScript' 
+for (const char of language) {
+    console.log(`character is : ${char}`)
+}
+```
+---
+hideInToc: true
+---
+
+- <span class='text-[#5D8392]'>For in loop</span>: It loops through the properties of an array.
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:542]: loops through properties of an object not array

File: pages/fundamentals.md [hunk 7]
```diff
@@ -65,6 +215,457 @@ let isMarried = confirm('Are you married?');
...
+
+let userName = 'Busola';
+console.log( greetUsers(userName));
+```
+
+---
+hideInToc: true
+---
+
+- Arror functions is a very simple and concise syntax for creating functions.
```
Comment (Oluwasetemi) [inline pages/fundamentals.md:656]: the arraow spelling.

**Review by Oluwasetemi — COMMENTED** [review]
Another very interesting thing that we can add to the function , function expression, arrow function and the one liner arrow function is to you `magic move` to show the transition from the function declaration to arrow function one liner.

```js
function sum(a, b) {
  return a + b
}
```

```js 
let sum = function add(a, b) { // complete this }
```

```js 
let sum = (a, b) => {
  return a + b
}
```

```js 
let sum = (a, b) => a + b
```

[https://sli.dev/features/shiki-magic-move](https://sli.dev/features/shiki-magic-move)

**Comment by Rolalove** [comment]
> Another very interesting thing that we can add to the function , function expression, arrow function and the one liner arrow function is to you `magic move` to show the transition from the function declaration to arrow function one liner.
> 
> ```js
> function sum(a, b) {
>   return a + b
> }
> ```
> 
> ```js
> let sum = function add(a, b) { // complete this }
> ```
> 
> ```js
> let sum = (a, b) => {
>   return a + b
> }
> ```
> 
> ```js
> let sum = (a, b) => a + b
> ```
> 
> https://sli.dev/features/shiki-magic-move


so I was thinking after the explanation of each type, i can use  the magic move for one example, explaining the three point

````md magic-move 

```js {monaco-run}
function greetUsers (name) {
    return 'Hello, ' + name + ' welcome to JavaScript course 101';
}
let userName = 'Busola'
let greetingMessage = greetUsers(userName);
console.log(greetingMessage)
```

```js 
const greetUsers = (name) => `Hello, ${name} welcome to JavaScript course 101`;

let userName = 'Busola';
console.log(greetUsers(userName))
```

```js
const greetUsers = (name) => `Hello, ${name} welcome to JavaScript course 101`;

let userName = 'Busola';
console.log(greetUsers(userName))
```

By doing this i can't show the output of each function, i don't know if this is the right direction.

**Comment by Oluwasetemi** [comment]
> ```
> function greetUsers (name) {
>     return 'Hello, ' + name + ' welcome to JavaScript course 101';
> }
> let userName = 'Busola'
> let greetingMessage = greetUsers(userName);
> console.log(greetingMessage)
> ```

done 👍

**Pull request merged**